finder-app/writer.c

```diff
@@ -0,0 +1,176 @@
+/*
+ *  References: 
+ *  https://vi.stackexchange.com/questions/7975/how-can-i-change-the-indent-size
+ *  https://www.geeksforgeeks.org/command-line-arguments-in-c-cpp
+ *  https://pubs.opengroup.org/onlinepubs/7908799/xsh/libgen.h.html
+ *  https://codeforwin.org/2018/03/c-program-check-file-or-directory-exists-not.html
+ *  https://pubs.opengroup.org/onlinepubs/009695299/functions/mkdir.html
+ *  https://pubs.opengroup.org/onlinepubs/007908799/xsh/sysstat.h.html
+ *  https://stackoverflow.com/questions/1658476/c-fopen-vs-open
+ *  https://linux.die.net/man/3/syslog
+ *  https://linux.die.net/man/8/syslogd
+ *  author: Rishab Shah
+ *  course: aesd - assignment 2
+*/
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <libgen.h>	    	/* dirname,basename,mkdir */
+#include <unistd.h>	    	/* access() */
+#include <sys/stat.h>		/* directory ecistence check */
+#include <errno.h>	    	/* perror() */
+#include <stdarg.h>		/* syslog functionality */
+#include <syslog.h>		/* for syslog open/close functioanlity */
+
+//function prototype
+int checkdirpathexistence(const char* path);
+int check_file_existence(const char *path);
+int create_file_and_write(const char* path, const char* string_to_write, const char* file_name);
+
+//Start of main program
+int main(int argc,char *argv[])
+{
+  openlog(NULL,0,LOG_USER);
+  
+  //Check for required number of arguments
+  if(argc != 3)
+  {
+#if 1
+    printf("ERROR: Invalid number of arguments\n");
+    printf("Total number of arguments should be 2\n");
+    printf("The order of the arguments should be:\n");
+    printf("[./writer] [arg 1] [arg 2]\n");
+    printf("  1)Name of the file along with the directory. Ex - /home/rishab/abcd.txt\n");
+    printf("  2)String to be written inside the file mentioned in arguement 1 i.e. inside abcd.txt\n"); 
+#endif
+    syslog(LOG_ERR,"ERROR: Invalid number of arguments\n");
+    syslog(LOG_ERR,"Total number of arguments should be 2\n");
+    syslog(LOG_ERR,"The order of the arguments should be:\n");
+    syslog(LOG_ERR,"[./writer] [arg 1] [arg 2]\n");
+    syslog(LOG_ERR,"  1)Name of the file along with the directory. Ex - /home/rishab/abcd.txt\n");
+    syslog(LOG_ERR,"  2)String to be written inside the file mentioned in arguement 1 i.e. inside abcd.txt\n"); 
```

Comment:
@Rishab-Shah since these prints are interactive you could just send them to stdout and not syslog.